Revise for ELF File Header section (#225)
riscv-non-isa/riscv-elf-psabi-doc: riscv-elf.adoc

=== File Header

The section below lists the defined RISC-V-specific values for several ELF
header fields; any fields not listed in this section have no RISC-V-specific
values.

e_ident::
  EI_CLASS::: Specifies the base ISA, either RV32 or RV64.
  Linking RV32 and RV64 code together is not supported.
+
--
[horizontal]
ELFCLASS64:::: ELF-64 Object File
ELFCLASS32:::: ELF-32 Object File
--
  EI_DATA::: Specifies the endianness; either big-endian or little-endian.
  Linking big-endian and little-endian code together is not supported.
+
--
[horizontal]
ELFDATA2LSB:::: Little-endian Object File
ELFDATA2MSB:::: Big-endian Object File
--

> +

Reviewer: Is this needed? Seems odd to me, but not an AsciiDoc expert.

Author: Updated, seems can be removed without changing layout

=== File Header

> The section below lists the defined RISC-V-specific values for several ELF
> header fields, field not listed in this section means those field didn't have
> RISC-V specific values.

Reviewer: ```suggestion
The section below lists the defined RISC-V-specific values for several ELF
header fields; any fields not listed in this section have no RISC-V-specific
values.
```

e_ident::

>   EI_CLASS::: Specifies the base ISA, either RV32 or RV64.  The linker disallow
>   linking RV32 and RV64 code together.

Reviewer: ```suggestion
  EI_CLASS::: Specifies the base ISA, either RV32 or RV64.
  Linking RV32 and RV64 code together is not supported.
```

+
--
[horizontal]
ELFCLASS64:::: ELF-64 Object File
ELFCLASS32:::: ELF-32 Object File
--

>   EI_DATA::: Specifies the endian, either big-endian or little-endian.
>   The linker disallow linking big-endian and little code together.

Reviewer: ```suggestion
  EI_DATA::: Specifies the endianness; either big-endian or little-endian.
  Linking big-endian and little-endian code together is not supported.
```

+
--
[horizontal]
ELFDATA2LSB:::: Little-endian Object File
ELFDATA2MSB:::: Big-endian Object File
--
+

> NOTE: RISC-V was only support little-endian, but RISC-V spec added big-endian

Reviewer: Does this commentary really belong here?

Author: My original intention is to explain why it should not fixed to `ELFDATA2LSB`, but actually it's kind of weird the describe here.

since https://github.com/riscv/riscv-isa-manual/commit/2151617320bc7d27d3ce8415be306ab51c362508[Jun, 2019].
+

>   EI_ABIVERSION::: Specifies the ABI version, this field have to be set to 0,

Reviewer: From the gABI:

> EI_ABIVERSION
Byte e_ident[EI_ABIVERSION] identifies the version of the ABI to which the object is targeted. This field is used to distinguish among incompatible versions of an ABI. The interpretation of this version number is dependent on the ABI identified by the EI_OSABI field. If no values are specified for the EI_OSABI field by the processor supplement or no version values are specified for the ABI determined by a particular value of the EI_OSABI byte, the value 0 shall be used for the EI_ABIVERSION byte; it indicates unspecified.

We therefore neither need to specify it nor should we as it can be OS-dependent

Author: Yeah, that seems should be bundle with `EI_OSABI`, but we are trying to avoid include OS-depended part in here, so let remove that.
